local/network.go

```diff
@@ -1099,17 +1109,23 @@ func (ln *localNetwork) buildFlags(
 		return buildFlagsReturn{}, err
 	}
 
-	// Use random free API port unless given in config file
-	apiPort, err := getPort(nodeConfig.Flags, configFile, config.HTTPPortKey)
-	if err != nil {
-		return buildFlagsReturn{}, err
+	// Prefer API port and P2P (staking) port from nodeConfig
+	apiPort, p2pPort := nodeConfig.ApiPort, nodeConfig.StakingPort
```

Comment:
What if those ports were already taken?

Reply:
The ports would be used by the same node previously, so it seems reasonable to assume they are available after the node shuts down?
Not sure if you already handled this is a different way but it is useful eg to not have to reconfigure metamask or other clients to submit tx after restarting the node.

Reply:
Yeah it is already handled in other way in ANR